aubreylib/resource.py

```diff
@@ -48,7 +48,7 @@ def get_desc_metadata(metadata_filename, metadata_type):
     """ Get the descriptive metadata for the object """
     # Open and read the metadata file into a StringIO filehandle
     metadata_filehandle = open_system_file(metadata_filename)
-    metadata_stringfile = StringIO.StringIO(metadata_filehandle.read())
+    metadata_stringfile = BytesIO(metadata_filehandle.read().encode('utf-8'))
```

Comment:
I could be wrong, but isn't this converting the output from `read()`, which I presume is Bytes, to String, then going _back_ to Bytes? _If_ that's the case, then I would think we could just do the following:
```suggestion
    metadata_stringfile = metadata_filehandle.read()
```

Reply:
But that could be wrong. Need to be sure before we would make such a change.

Reply:
In Python 3, the way things are in `open_system_file`, that read() could return `bytes` or `str` depending if the file is being read with `urlopen` (now bytes) or from the file system with `open` (by default assumes text and returns a `str`). We may want to modify `aubreylib.system.open_system_file` to return a consistent type.

Reply:
I can make the `aubreylib.system.open_system_file` function return bytes type and then apply BytesIO on the filehandler, which will be consumed by `pyuntl.untldoc.untlxml2pydict` function.

Reply:
I think `open_system_file` used to always return `str` in Python 2, I haven't looked at all of this, but did you make sure we want to be dealing with `bytes` rather than `str` now?

Reply:
The untlxml2pydict() makes a call to lxml.etree.iterparse(file_name), which according to my understanding should either read the file directly or get the data read as bytes object.